packages/next/build/swc/src/hook_optimizer.rs

```diff
@@ -0,0 +1,120 @@
+use swc_common::DUMMY_SP;
+use swc_ecmascript::ast::{
+  ArrayPat, Decl, Expr, ExprOrSuper, Ident, ImportDecl, ImportSpecifier, KeyValuePatProp, Number,
+  ObjectPat, ObjectPatProp, Pat, PropName, VarDecl, VarDeclarator,
+};
+use swc_ecmascript::visit::{Fold, FoldWith};
+
+pub fn hook_optimizer() -> impl Fold {
+  HookOptimizer::default()
+}
+
+#[derive(Debug, Default)]
+struct HookOptimizer {
+  hooks: Vec<String>,
+}
+
+impl Fold for HookOptimizer {
+  // Find hooks imported from react/preact
+  fn fold_import_decl(&mut self, decl: ImportDecl) -> ImportDecl {
+    let ImportDecl {
+      ref src,
+      ref specifiers,
+      ..
+    } = decl;
+    if &src.value == "react" || &src.value == "preact/hooks" {
+      for specifier in specifiers {
+        if let ImportSpecifier::Named(named_specifier) = specifier {
+          if String::from(&named_specifier.local.sym as &str).starts_with("use") {
```

Comment:
This can be replaced with `named_specifier.local.sym.starts_with("use")`.
Not so important as I expect `rustc` to optimize it.